src/plugins/explore/public/application/utils/languages/types.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Query } from '../../../../../data/common';
+
+export type QueryWithQueryAsString = Omit<Query, 'query'> & {
+  query: string;
+};
```

Comment:
maybe also update https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b5882e98bea143d921c789372ce6b38b0b65aeae/src/plugins/query_enhancements/common/utils.ts#L147-L147
to use this type

Reply:
i think this type should be defined next to Query (i think its defined in data plugin?) , but i didn't want to touch any code outside explore plugin. we can do that later